src/Init/Data/Repr.lean

```diff
@@ -13,11 +13,24 @@ open Sum Subtype Nat
 
 open Std
 
+/--
+A typeclass that specifies the standard way of turning values of some type into `Format`.
+
+When rendered this `Format` should be as close as possible to something that can be parsed as the
+input value.
+-/
 class Repr (α : Type u) where
+  /--
+  Turn a value of type `α` into `Format` at a given precedence. The precedence value can be used
+  to avoid parenthesis if they are not necessary.
```

Comment:
```suggestion
  to avoid parentheses if they are not necessary.
```
Where can the user go to find out what the reasonable values of the `Nat` are, and how they use this `Nat` in practice to add parentheses? It might also be good to refer to useful helpers here, like `addAppParen`.

Reply:
I don't think this Nat is used very much, it's nothing like the global precedence system used with `notation` et al. Most of the uses are very local and I've only ever seen 1 or 2 different precedences used.